TELEGRAF-3: Wireless stats input plugins for HiveOS

chayan-04 opened the pull request:
Creating an input plugin to retrieve wireless stats [rf stats and client stats] from driver and send
to http server/output plugins.

chayan-04 committed:
TELEGRAF-3: Wireless stats input plugins for HiveOS

Creating an input plugin to retrieve wireless stats
[rf stats and client stats] from driver and send
to http server/output plugins.

arnelplim commented on plugins/inputs/all/all.go:
```diff
@@ -208,4 +208,5 @@ import (
 // 	_ "github.com/influxdata/telegraf/plugins/inputs/zfs"
 // 	_ "github.com/influxdata/telegraf/plugins/inputs/zipkin"
 // 	_ "github.com/influxdata/telegraf/plugins/inputs/zookeeper"
+	_ "github.com/influxdata/telegraf/plugins/inputs/ah_wireless"
```
How much does this plugin increase the size of the executable?  We have size constraints on our EXOS summitlite_arm platform.

chayan-04 replied:
@arnelplim 
Without plugin 13039575
With plugin 13222735

So its almost 200Kb of increment.
Is that ok? How much is the size constraint?

arnelplim replied:
200kb might be a problem.  EXOS and VOSS doesn't need this plugin, so can we have it commented out by default and adjust your build adjust to include it for wireless platform only?

arnelplim replied:
@vullanatt Thoughts?  I think 200k will be an issue on summitlite_arm, right?

chayan-04 replied:
> 200kb might be a problem. EXOS and VOSS doesn't need this plugin, so can we have it commented out by default and adjust your build adjust to include it for wireless platform only?

This makes sense since, this plugin wont work in any other platform apart from hive

chayan-04 replied:
Disabled the plugin by default.

arnelplim commented on plugins/inputs/ah_wireless/ah_wireless.go:
```diff
@@ -0,0 +1,942 @@
+package ah_wireless
+
```
Will you be providing tests?  i.e. ah_wireless_test.go

arnelplim replied:
Same question with README.md file.

chayan-04 replied:
@arnelplim , I will provide test and readme once all the features are incorporated. This is initial commit. We shall have incremental commits.

arnelplim commented:
extr_version needs to be incremented

zhchen-extr commented on plugins/inputs/ah_wireless/ah_wireless.go:
```diff
@@ -0,0 +1,942 @@
+}
+
+
+func wl_ioctl(fd int, op, argp uintptr) error {
```
wl_ioctl is the same as fe_ioctl. Consider unifying them as one function.

chayan-04 replied:
@zhchen-extr , Yes. That was a good point. I have changed it now.

arnelplim commented:
Added Marian who handles Telegraf on FabricEngine (VOSS) platform.

chayan-04 committed:
TELEGRAF-3: Code cleanup and optimization

Cleaning up unused and commented lines
Also using a single ioctl function.
Disabling the plugin so that other platforms
are not affected.
